### PR TITLE
feat: add a focusTriggerAfterClose prop to Modal for passing down to rc-dialog

### PR DESCRIPTION
**Pull request by molokovev**
Also:
- set default value to true, similarly to rc-dialog
- add comments to the Modal docs. CN is google-translated though.

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

I've faced an issue when after selecting something in a Modal I need to close it and to update a global state and then an input is supposed to reflect to redux action and being focused. But the antd's Modal focuses back the modal activating button after hide animation ends.
That's why this props would help. Actually, this prop can be set right now but without TS support though.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | `focusTriggerAfterClose` prop is added to Modal props type definitions |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/modal/index.en-US.md](https://github.com/molokovev/ant-design/blob/pr/add_missing_modal_focusTriggerAfterClose_prop_type/components/modal/index.en-US.md)
[View rendered components/modal/index.zh-CN.md](https://github.com/molokovev/ant-design/blob/pr/add_missing_modal_focusTriggerAfterClose_prop_type/components/modal/index.zh-CN.md)

### CODE REVIEW

**Comment by afc163** [comment]
feature branch

**Comment by molokovev** [comment]
> feature branch

**Comment by molokovev** [comment]
> feature branch

Hi! Right, I wasn't sure if typings fix is either a bugfix or a feature. Thanks, fixed.

**Comment by afc163** [comment]
Other commits is contained.

**Comment by afc163** [comment]
Other commits is contained.

image

**Comment by zombieJ** [comment]
@molokovev PR contains other commit. Pls help to clean up.

**Comment by molokovev** [comment]
Removed other commits

File: components/modal/index.zh-CN.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/3StSdUlSH/Modal.svg
 | destroyOnClose | 关闭时销毁 Modal 里的子元素 | boolean | false |  |
 | footer | 底部内容，当不需要默认底部按钮时，可以设为 `footer={null}` | ReactNode | (确定取消按钮) |  |
 | forceRender | 强制渲染 Modal | boolean | false |  |
+| focusTriggerAfterClose | 对话框关闭后是否需要聚焦触发元素 | boolean | true |  |
```
Comment (afc163) [inline components/modal/index.zh-CN.md:32]:
```suggestion
| focusTriggerAfterClose | 对话框关闭后是否需要聚焦触发元素 | boolean | true | 4.9.0 |
```

File: components/modal/index.en-US.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ When requiring users to interact with the application, but without jumping to a
 | destroyOnClose | Whether to unmount child components on onClose | boolean | false |  |
 | footer | Footer content, set as `footer={null}` when you don't need default buttons | ReactNode | (OK and Cancel buttons) |  |
 | forceRender | Force render Modal | boolean | false |  |
+| focusTriggerAfterClose | Whether need to focus trigger element after dialog is closed | boolean | true |  |
```
Comment (afc163) [inline components/modal/index.en-US.md:29]:
```suggestion
| focusTriggerAfterClose | Whether need to focus trigger element after dialog is closed | boolean | true | 4.9.0 |
```

**Review by afc163 — APPROVED** [review]

**Pull request merged**